src/System.DirectoryServices/src/System/DirectoryServices/SearchResultCollection.cs

```diff
@@ -422,14 +422,11 @@ private void CleanLastError()
                 SafeNativeMethods.ADsSetLastError(SafeNativeMethods.ERROR_SUCCESS, null, null);
             }
 
-            private int GetLastError(ref int errorCode)
+            private unsafe int GetLastError(ref int errorCode)
             {
-                StringBuilder errorBuffer = new StringBuilder();
-                StringBuilder nameBuffer = new StringBuilder();
+                char c1 = '\0', c2 = '\0';
                 errorCode = SafeNativeMethods.ERROR_SUCCESS;
-                int hr = SafeNativeMethods.ADsGetLastError(out errorCode, errorBuffer, 0, nameBuffer, 0);
-
-                return hr;
+                return SafeNativeMethods.ADsGetLastError(out errorCode, &c1, 0, &c2, 0);
```

Comment:
>0 [](start = 77, length = 1)

it is weird we are passing 0 here. I know the original code were doing that but the documentation saying 

"Size, in characters, of the lpErrorBuf buffer. If the buffer is too small to receive the error string, the string is truncated, but still null-terminated. A buffer, of at least 256 bytes, is recommended"

so it store the null anyway even if we saying the buffer length is 0?

Reply:
Yeah, I agree it's odd.  I just wanted to keep things as close to the data we're currently passing in as possible.